File: Realidad Virtual/1 Primer Momento/Tarea 1/Main(1).cs
public class Movimiento
{
    public int ID { get; set; }
    public double EjeX { get; set; }
    public double EjeY { get; set; }
    public double EjeZ { get; set; }
    public DateTime Fecha { get; set; }

    public Movimiento(int id, double ejeX, double ejeY, double ejeZ, DateTime fecha)
    {
        ID = id;
        EjeX = ejeX;
        EjeY = ejeY;
        EjeZ = ejeZ;
        Fecha = fecha;
    }

    public override string ToString()
    {
        return $"ID: {ID}, EjeX: {EjeX}, EjeY: {EjeY}, EjeZ: {EjeZ}, Fecha: {Fecha}";
    }
}

public class MovimientoManager
{
    private List<Movimiento> movimientos = new List<Movimiento>();
    private int nextId = 1;

    // Crear un nuevo movimiento
    public void CrearMovimiento(double ejeX, double ejeY, double ejeZ, DateTime fecha)
    {
        var movimiento = new Movimiento(nextId++, ejeX, ejeY, ejeZ, fecha);
        movimientos.Add(movimiento);
        Console.WriteLine("Movimiento creado: " + movimiento);
    }

    // Leer (listar) todos los movimientos
    public void ListarMovimientos()
    {
        if (movimientos.Count == 0)
        {
            Console.WriteLine("No hay movimientos disponibles.");
            return;
        }

        foreach (var movimiento in movimientos)
        {
            Console.WriteLine(movimiento);
        }
    }

    // Actualizar un movimiento existente
    public void ActualizarMovimiento(int id, double ejeX, double ejeY, double ejeZ, DateTime fecha)
    {
        var movimiento = movimientos.FirstOrDefault(m => m.ID == id);
        if (movimiento != null)
        {
            movimiento.EjeX = ejeX;
            movimiento.EjeY = ejeY;
            movimiento.EjeZ = ejeZ;
            movimiento.Fecha = fecha;
            Console.WriteLine("Movimiento actualizado: " + movimiento);
        }
        else
        {
            Console.WriteLine("Movimiento no encontrado.");
        }
    }

    // Eliminar un movimiento
    public void EliminarMovimiento(int id)
    {
        var movimiento = movimientos.FirstOrDefault(m => m.ID == id);
        if (movimiento != null)
        {
            movimientos.Remove(movimiento);
            Console.WriteLine("Movimiento eliminado: " + movimiento);
        }
        else
        {
            Console.WriteLine("Movimiento no encontrado.");
        }
    }
}

class Program
{
    static void Main(string[] args)
    {
        var manager = new MovimientoManager();

        while (true)
        {
            Console.WriteLine("\nGestión de Movimientos:");
            Console.WriteLine("1. Crear Movimiento");
            Console.WriteLine("2. Listar Movimientos");
            Console.WriteLine("3. Actualizar Movimiento");
            Console.WriteLine("4. Eliminar Movimiento");
            Console.WriteLine("5. Salir");
            Console.Write("Seleccione una opción: ");
            var opcion = Console.ReadLine();

            switch (opcion)
            {
                case "1":
                    Console.Write("Eje X: ");
                    var ejeX = double.Parse(Console.ReadLine());
                    Console.Write("Eje Y: ");
                    var ejeY = double.Parse(Console.ReadLine());
                    Console.Write("Eje Z: ");
                    var ejeZ = double.Parse(Console.ReadLine());
                    Console.Write("Fecha (YYYY-MM-DD): ");
                    var fecha = DateTime.Parse(Console.ReadLine());
                    manager.CrearMovimiento(ejeX, ejeY, ejeZ, fecha);
                    break;
                case "2":
                    manager.ListarMovimientos();
                    break;
                case "3":
                    Console.Write("ID del Movimiento a actualizar: ");
                    var idActualizar = int.Parse(Console.ReadLine());
                    Console.Write("Nuevo Eje X: ");
                    ejeX = double.Parse(Console.ReadLine());
                    Console.Write("Nuevo Eje Y: ");
                    ejeY = double.Parse(Console.ReadLine());
                    Console.Write("Nuevo Eje Z: ");
                    ejeZ = double.Parse(Console.ReadLine());
                    Console.Write("Nueva Fecha (YYYY-MM-DD): ");
                    fecha = DateTime.Parse(Console.ReadLine());
                    manager.ActualizarMovimiento(idActualizar, ejeX, ejeY, ejeZ, fecha);
                    break;
                case "4":
                    Console.Write("ID del Movimiento a eliminar: ");
                    var idEliminar = int.Parse(Console.ReadLine());
                    manager.EliminarMovimiento(idEliminar);
                    break;
                case "5":
                    return;
                default:
                    Console.WriteLine("Opción no válida.");
                    break;
            }
        }
    }
}

File: Realidad Virtual/1 Primer Momento/Tarea 1/main(1).cpp
#include <iostream>
#include <list>
#include <string>

using namespace std;

class Puntaje {
public:
    int id;
    string nombre;
    int puntos;

    Puntaje(int id, string nombre, int puntos) : id(id), nombre(nombre), puntos(puntos) {}
};

class CRUD {
private:
    list<Puntaje> puntajes;

public:
    void crearPuntaje(int id, string nombre, int puntos) {
        puntajes.push_back(Puntaje(id, nombre, puntos));
    }

    void leerPuntajes() {
        for (const auto& puntaje : puntajes) {
            cout << "ID: " << puntaje.id << ", Nombre: " << puntaje.nombre << ", Puntos: " << puntaje.puntos << endl;
        }
    }

    void actualizarPuntaje(int id, string nombre, int puntos) {
        for (auto& pnt : puntajes) {
            if (pnt.id == id) {
                pnt.nombre = nombre;
                pnt.puntos = puntos;
                return;
            }
        }
        cout << "Puntaje con ID " << id << " no encontrado." << endl;
    }

    void eliminarPuntaje(int id) {
        for (auto it = puntajes.begin(); it != puntajes.end(); ++it) {
            if (it->id == id) {
                puntajes.erase(it);
                return;
            }
        }
        cout << "Puntaje con ID " << id << " no encontrado." << endl;
    }
};

int main() {
    CRUD crud;

    int opcion;
    while (true) {
        cout << "\n1. Crear puntaje\n2. Leer puntajes\n3. Actualizar puntaje\n4. Eliminar puntaje\n5. Salir\n";
        cout << "Selecciona una opción: ";
        cin >> opcion;

        if (opcion == 5) break;

        int id;
        string nombre;
        int puntos;

        switch (opcion) {
            case 1:
                cout << "Ingrese ID: ";
                cin >> id;
                cout << "Ingrese nombre: ";
                cin >> nombre;
                cout << "Ingrese puntos: ";
                cin >> puntos;
                crud.crearPuntaje(id, nombre, puntos);
                break;
            case 2:
                crud.leerPuntajes();
                break;
            case 3:
                cout << "Ingrese ID del puntaje a actualizar: ";
                cin >> id;
                cout << "Ingrese nuevo nombre: ";
                cin >> nombre;
                cout << "Ingrese nuevos puntos: ";
                cin >> puntos;
                crud.actualizarPuntaje(id, nombre, puntos);
                break;
            case 4:
                cout << "Ingrese ID del puntaje a eliminar: ";
                cin >> id;
                crud.eliminarPuntaje(id);
                break;
            default:
                cout << "Opción no válida." << endl;
                break;
        }
    }

    return 0;
}
